### PR TITLE
WP Ai Global Styles

### PR DESCRIPTION
**Pull request by GunkaArtur**
Related issue https://github.com/bagrinsergiu/blox-editor/issues/25691

### CODE REVIEW

**Comment by GunkaArtur** [comment]
URL from config will be added here https://github.com/bagrinsergiu/blox-editor/issues/26071

File: public/editor-client/src/aiGlobalStyles/index.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { t } from "@/utils/i18n";
+import { AiGlobalStyles, FontStyle, Palette } from "../types/DefaultTemplate";
+
+export const getAIGlobalStyles = (): AiGlobalStyles<Palette, FontStyle> => {
+  const apiUrl =
+    "https://phplaravel-1109775-4163586.cloudwaysapps.com/api/template";
```
Comment (victor-pg) [inline public/editor-client/src/aiGlobalStyles/index.ts:6]: use url from config

File: public/editor-client/src/aiGlobalStyles/index.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  return {
+    async getColors(res, rej) {
+      try {
+        const styles = await fetch(`${apiUrl}/style`).then((r) => r.json());
```
Comment (victor-pg) [inline public/editor-client/src/aiGlobalStyles/index.ts:11]: use async/await without `then`:

```ts
const styles = await getStyles(/* ... */)
```

File: public/editor-client/src/aiGlobalStyles/index.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+        res(styles.colorPalette);
+      } catch (e) {
+        rej(t("Failed to load meta.json"));
+      }
+    },
+    async getTypography(res, rej) {
+      try {
+        const data = await fetch(`${apiUrl}/typography`).then((r) => r.json());
```
Comment (victor-pg) [inline public/editor-client/src/aiGlobalStyles/index.ts:20]: here also use async/await without `then`

**Review by victor-pg — CHANGES_REQUESTED** [review]

File: public/editor-client/src/aiGlobalStyles/index.ts (hunk 1 above)
Comment (GunkaArtur) [reply public/editor-client/src/aiGlobalStyles/index.ts:6]: https://github.com/bagrinsergiu/blox-editor/issues/26071

File: editor/editor/editor.php [hunk 4]
```diff
@@ -93,6 +93,7 @@ public function getClientConfig($context)
             'project' => array(
                 'status' => $this->getProjectStatus(),
             ),
+	        'aiGlobalStyleUrl' => 'https://phplaravel-1109775-4163586.cloudwaysapps.com'
```
Comment (maxval1) [inline editor/editor/editor.php:96]: need to be revied

File: public/editor-client/src/aiGlobalStyles/index.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+export const getAIGlobalStyles = (
+  config: Config
+): AiGlobalStyles<Palette, FontStyle> => {
+  const { aiGlobalStyleUrl } = config;
+
+  return {
+    async getColors(res, rej) {
+      try {
+        const styles = await fetch(
+          `${aiGlobalStyleUrl}/api/template/style`
```
Comment (maxval1) [inline public/editor-client/src/aiGlobalStyles/index.ts:14]: same as editor api client

**Review by maxval1 — CHANGES_REQUESTED** [review]

File: public/editor-client/src/aiGlobalStyles/index.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  config: Config
+): AiGlobalStyles<Palette, FontStyle> => {
+  return {
+    async regenerateColors(res, rej) {
+      try {
+        const styles = await getStyles(config);
+
+        res(styles.colorPalette);
+      } catch (e) {
+        rej(t("Failed to load meta.json"));
```
Comment (victor-pg) [inline public/editor-client/src/aiGlobalStyles/index.ts:16]: review reject message

File: public/editor-client/src/aiGlobalStyles/index.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      }
+    },
+    async regenerateTypography(res, rej) {
+      try {
+        const data = await getTypography(config);
+
+        res(data.fontStyles);
+      } catch (e) {
+        rej(t("Failed to load resolves for selected DefaultTemplate"));
```
Comment (victor-pg) [inline public/editor-client/src/aiGlobalStyles/index.ts:25]: review reject message

shouldn't be something like: "failed to load typography..." ?

File: public/editor-client/src/types/DefaultTemplate.ts [hunk 8]
```diff
@@ -5,6 +5,15 @@ export interface DefaultBlock {
   value: Record<string, unknown>;
 }
 
+export interface AiGlobalStyles<T1, T2> {
```
Comment (victor-pg) [inline public/editor-client/src/types/DefaultTemplate.ts:8]: rename `AiGlobalStyles`

as i see in previous @maxval1 review there is nothing with AI

File: public/editor-client/src/index.ts [hunk 9]
```diff
@@ -90,6 +91,11 @@ if (window.__VISUAL_CONFIG__) {
   // UI
   if (window.__VISUAL_CONFIG__.ui) {
     window.__VISUAL_CONFIG__.ui.publish = publish;
+    set(
+      window.__VISUAL_CONFIG__.ui,
+      ["leftSidebar", "styles"],
+      getAIGlobalStyles(config)
```
Comment (victor-pg) [inline public/editor-client/src/index.ts:97]: rename also this function

**Review by victor-pg — CHANGES_REQUESTED** [review]

**Review by lift46252 — APPROVED** [review]

**Pull request merged**